src/test/ui/async-await/async-fn-size-moved-locals.rs

```diff
@@ -9,6 +9,7 @@
 
 // ignore-emscripten (sizes don't match)
 // run-pass
+// needs-unwind Async futures off by one byte due to panic byte flag required
```

Comment:
It looks like we can use `#[cfg(panic = "abort")]` to change the checks for unwind/abort, I didn't know that was a thing previously. Instead, we should consider a pattern like:

```rust
#[cfg(panic = "unwind")]
const SINGLE_SIZE: usize = 1025;
#[cfg(panic = "abort")]
const SINGLE_SIZE: usize = 1024;

assert_eq!(SINGLE_SIZE, std::mem::size_of_val(&single()));
```

This will make sure that our sizes are tested for both unwind and abort and remain stable over time.

Reply:
For this test, only 2 fail on `abort`:

```
assert_eq!(1026, std::mem::size_of_val(&single_with_noop())); # unwind
assert_eq!(1025, std::mem::size_of_val(&single_with_noop())); # abort

assert_eq!(6157, std::mem::size_of_val(&mixed_sizes())); # unwind
assert_eq!(6153, std::mem::size_of_val(&mixed_sizes())); # abort
```

I'm not seeing any patterns amongst these failure cases and the ones that pass, and am still troubleshooting. Making this comment in case anything is obvious from these changes!